code/modules/antagonists/horror/horror_mutate.dm

```diff
@@ -0,0 +1,97 @@
+// Horror mutation menu
+// Totally not a copypaste of darkspawn menu, not a copypaste of cellular emporium, i swear. Edit: now looks like guardianbuilder too
+
+/mob/living/simple_animal/horror/proc/has_ability(typepath)
+	for(var/datum/action/innate/horror/ability in horrorabilities)
+		if(istype(ability, typepath))
+			return ability
+	return
+
+/mob/living/simple_animal/horror/proc/add_ability(typepath)
+	if(has_ability(typepath))
+		return
+	var/datum/action/innate/horror/action = new typepath
+	action.B = src
+	horrorabilities += action
+	RefreshAbilities()
+	to_chat(src, span_velvet("You have mutated the <b>[action.name]</b>."))
+	available_points = max(0, available_points - action.soul_price)
+	return TRUE
+
+/mob/living/simple_animal/horror/proc/has_upgrade(id)

```

Comment:
You're not using this in the rest of the code?

Reply:
i am, it's used for stuff like invisibility upon exiting body

Reply:
Unless I did something wrong I only see this proc used 3 times
For the rest you’re just checking the list directly?

Reply:
oh yeah i'll replace the checking list directly, i'm fucking tired

Reply:
> oh yeah i'll replace the checking list directly, i'm fucking tired

i can relate lmao